### PR TITLE
[202012] Stop PMON before stopping BGP while doing warmboot/fastboot

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
#### Why I did it

This is the PR for 202012 only as https://github.com/Azure/sonic-buildimage/pull/10046 cannot be cherry-picked to 202012.

Stopping swss and syncd causes some driver module unloading. Those driver modules are depended by PMON. This could trigger ERROR logs in syslog like:

```
Failed to run thermal policy all fan and psu presence
```

#### How I did it

Adjust warmboot shutdown order in make file

#### How to verify it

Manual test

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

**Review by liat-grozovik — APPROVED** [review]

**Comment by vaibhavhd** [comment]
Please add some description to the PR. Specifically what kind of error logs appear if `pmon` is stopped after BGP stop.
Also, I think this change can be generic, so mellanox check can be removed.

**Comment by vaibhavhd** [comment]
This change should also be added to Master branch, and preferably cherry picked from there into 202012.

**Comment by liat-grozovik** [comment]
@vaibhavhd sure. I marked it ready for review before @Junchao-Mellanox added the description. My fault.

**Comment by Junchao-Mellanox** [comment]
@vaibhavhd added description.

**Review by liat-grozovik — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@Junchao-Mellanox  to be able to merge it the branch should be up to date. Can you please take care?

**Comment by Junchao-Mellanox** [comment]
Hi Liat, it is already the latest.

File: scripts/fast-reboot [hunk 1]
```diff
@@ -612,6 +612,13 @@ debug "Stopping radv service..."
 systemctl stop radv
 debug "Stopped radv service..."
 
+# Kill pmon before stopping BGP/teamd/swss/syncd to prevent error log
+if [[ "x$sonic_asic_type" == x"mellanox" ]]; then
```
[the remark this answers is not in the document]
Comment (Junchao-Mellanox) [reply scripts/fast-reboot:616]: Hi @yxieca , much appreciate if you could provide your comment here.

[the remark this answers is not in the document]
Comment (yxieca) [reply scripts/fast-reboot:616]: Is there specific reason we only want to do this for Mellanox platform? I agree with Vaibhav if there is no particular reason to only do it for Mellanox platform.

[the remark this answers is not in the document]
Comment (Junchao-Mellanox) [reply scripts/fast-reboot:616]: I don't have chance to do test on other platform so I only do the change for Mellanox platform.

[the remark this answers is not in the document]
Comment (vaibhavhd) [reply scripts/fast-reboot:616]: I can run a test on non-mellanox platform. I will update here soon.

[the remark this answers is not in the document]
Comment (vaibhavhd) [reply scripts/fast-reboot]: I still see these messages when shutting down pmon before bgp. Do you see them too?

But these don't seem to have caused any detrimental impact. Also, this is still useful as preempting pmon shutdown saves time after syncd shutdown.
```
Mar 23 21:19:28.859057 str-s6100-acs-4 NOTICE admin: Stopping pmon ...

Mar 23 21:19:29.365987 str-s6100-acs-4 DEBUG /container: container_stop: BEGIN
Mar 23 21:19:29.367554 str-s6100-acs-4 DEBUG /container: read_data: config:True feature:pmon fields:[('set_owner', 'local'), ('no_fallback_to_local', False)] val:['kube', False]
Mar 23 21:19:29.369057 str-s6100-acs-4 DEBUG /container: read_data: config:False feature:pmon fields:[('current_owner', 'none'), ('remote_state', 'none'), ('container_id', '')] val:['local', 'none', 'pmon']
Mar 23 21:19:29.370746 str-s6100-acs-4 DEBUG /container: container_stop: pmon: set_owner:kube current_owner:local remote_state:none docker_id:pmon
Mar 23 21:19:29.373080 str-s6100-acs-4 DEBUG /ctrmgrd.py: run: Received message : '('SET', 'SET', (('acms_enabled', 'false'), ('restapi_enabled', 'false'), ('bgp_enabled', 'false'), ('pmon_enabled', 'false'), ('pmon_20201231.57_enabled', 'false'), ('swss_enabled', 'false'), ('teamd_enabled', 'false'), ('syncd_enabled', 'false'), ('dhcp_relay_enabled', 'true'), ('lldp_enabled', 'true'), ('radv_enabled', 'true'), ('radv_20201231.57_enabled', 'false'), ('dhcp_relay_20201231.57_enabled', 'false'), ('lldp_20201231.57_enabled', 'false'), ('telemetry_enabled', 'false'), ('snmp_enabled', 'true'), ('snmp_20201231.57_enabled', 'false'), ('telemetry_20201231.57_enabled', 'false')))'
Mar 23 21:19:29.373282 str-s6100-acs-4 DEBUG /ctrmgrd.py: on_update: Skip label update: connected:False pending:False
Mar 23 21:19:30.296185 str-s6100-acs-4 INFO pmon#supervisord 2022-03-23 21:19:30,295 WARN received SIGTERM indicating exit request
Mar 23 21:19:30.297379 str-s6100-acs-4 INFO pmon#supervisord 2022-03-23 21:19:30,295 INFO waiting for supervisor-proc-exit-listener, rsyslogd, xcvrd, psud, syseepromd, thermalctld, pcied to die
Mar 23 21:19:31.346045 str-s6100-acs-4 INFO pmon#supervisord 2022-03-23 21:19:31,345 INFO stopped: pcied (exit status 143)
Mar 23 21:19:31.346213 str-s6100-acs-4 INFO pmon#thermalctld: Caught signal 'SIGTERM' - exiting...
Mar 23 21:19:31.445607 str-s6100-acs-4 INFO pmon#/supervisord: thermalctld Traceback (most recent call last):
Mar 23 21:19:31.445607 str-s6100-acs-4 INFO pmon#/supervisord: thermalctld   File "/usr/local/bin/thermalctld", line 882, in <module>
Mar 23 21:19:31.445607 str-s6100-acs-4 INFO pmon#/supervisord: thermalctld     sys.exit(main())
Mar 23 21:19:31.445681 str-s6100-acs-4 INFO pmon#/supervisord: thermalctld   File "/usr/local/bin/thermalctld", line 871, in main
Mar 23 21:19:31.445681 str-s6100-acs-4 INFO pmon#/supervisord: thermalctld     while thermal_control.run():
Mar 23 21:19:31.445681 str-s6100-acs-4 INFO pmon#/supervisord: thermalctld   File "/usr/local/bin/thermalctld", line 839, in run
Mar 23 21:19:31.445710 str-s6100-acs-4 INFO pmon#/supervisord: thermalctld     if self.stop_event.wait(self.wait_time):
Mar 23 21:19:31.445758 str-s6100-acs-4 INFO pmon#/supervisord: thermalctld   File "/usr/lib/python3.7/threading.py", line 552, in wait
Mar 23 21:19:31.445758 str-s6100-acs-4 INFO pmon#/supervisord: thermalctld     signaled = self._cond.wait(timeout)
Mar 23 21:19:31.445785 str-s6100-acs-4 INFO pmon#/supervisord: thermalctld   File "/usr/lib/python3.7/threading.py", line 300, in wait
Mar 23 21:19:31.445827 str-s6100-acs-4 INFO pmon#/supervisord: thermalctld     gotit = waiter.acquire(True, timeout)
Mar 23 21:19:31.445827 str-s6100-acs-4 INFO pmon#/supervisord: thermalctld   File "/usr/local/bin/thermalctld", line 826, in signal_handler
Mar 23 21:19:31.445853 str-s6100-acs-4 INFO pmon#/supervisord: thermalctld     self.thermal_manager.stop()
Mar 23 21:19:31.445853 str-s6100-acs-4 INFO pmon#/supervisord: thermalctld AttributeError: 'NoneType' object has no attribute 'stop'
Mar 23 21:19:31.491821 str-s6100-acs-4 INFO pmon#supervisord 2022-03-23 21:19:31,491 INFO stopped: thermalctld (exit status 1)
Mar 23 21:19:31.492507 str-s6100-acs-4 INFO pmon#syseepromd[50]: Caught signal 'SIGTERM' - exiting...
Mar 23 21:19:31.492853 str-s6100-acs-4 INFO pmon#syseepromd[50]: Shutting down...
```

**Review by vaibhavhd — APPROVED** [review]
LGTM

**Pull request merged**